sdk/scheduler/src/main/java/com/mesosphere/sdk/http/EndpointUtils.java

```diff
@@ -42,24 +46,31 @@ public static String toEndpoint(String hostname, int port) {
     /**
      * Returns the correct DNS domain for tasks within the service.
      */
-    public static String toAutoIpDomain(String serviceName) {
+    public static String toAutoIpDomain(String serviceName, SchedulerConfig schedulerConfig) {
+        Optional<String> customTLD = schedulerConfig.getCustomServiceTLD();
```

Comment:
Would it make sense to move this logic into the `SchedulerConfig` instance? i.e that we have a `SchedulerConfig.getTLD()` method that returns the default or the custom one.

Reply:
It would split the two kinds of TLDs (non-vip and vip).

VIPs will never work outside of a cluster so it doesn't make sense to support a custom TLD for them.

Reply:
I don't quite see how this has anything to do with `VIPs`. The only use of `Constants.DNS_TLD`  in this class is here. This is also the only place where we call the `getCustomServiceTLD` method.

Reply:
Ah, I wasn't getting what ya'll were putting down.